controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -397,7 +394,7 @@ func (in *KubeadmControlPlane) validateEtcd(prev *KubeadmControlPlane) (allErrs
 }
 
 func (in *KubeadmControlPlane) validateVersion(previousVersion string) (allErrs field.ErrorList) {
-	fromVersion, err := util.ParseMajorMinorPatch(previousVersion)
+	fromVersion, err := version.ParseMajorMinorPatch(previousVersion)
```

Comment:
should I makes this tolerant so it's strictly equivalent to previous behavior @vincepri ? This is validating k8s version which we enforce the "v" prefix for above so technically it shouldn't be tolerant but we were using the util helper before which is tolerant 😕

Reply:
More inclined to do the right thing, rather than keeping the backward compatible behavior. Conversion webhook to v1alpha4 can workaround this behavior when up-converting (given that we'll more stricter now)

Reply:
Sounds good. We're technically not more strict since we were already enforcing the "v" prefix with the semver in validate create in alpha3.